rotkehlchen/data_migrations/migrations/migration_20.py

```diff
@@ -145,4 +157,130 @@ def _fix_swap_event_identifiers(rotki: 'Rotkehlchen') -> None:
                     updates_to_apply,
                 )
 
+    @progress_step(description='Recovering lost trades from v1.39.0 upgrade')
+    def _recover_lost_trades_from_v139_upgrade(rotki: 'Rotkehlchen') -> None:
+        """Recover trades that were lost during the v1.38.4 to v1.39.0 database upgrade.
+
+        During the upgrade, trades with the same location and empty link field had ID conflicts.
+        Only the first trade was kept, while subsequent trades were silently dropped.
+        This function attempts to recover those lost trades from a database backup.
+        """
+        db = rotki.data.db
+        log.debug('checking for trades lost during v1.39.0 upgrade')
+
+        # check if any events have location hash as their identifier
+        # this indicates the upgrade bug affected this database
+        location_hashes = tuple(hash_id(str(loc)) for loc in Location)
+        placeholders = ','.join('?' * len(location_hashes))
+        with db.conn.read_ctx() as cursor:
+            if len(affected_events := cursor.execute(
+                f'SELECT event_identifier FROM history_events WHERE event_identifier IN ({placeholders})',  # noqa: E501
+                location_hashes,
+            ).fetchall()) == 0:
```

Comment:
why not do a count instead of len? YOu are always reading len and not the actual event_identifiers for what I can see